packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutWithAddCardPanel/DashboardGridLayoutWithAddCardPanel.tsx

```diff
@@ -64,66 +75,139 @@ export class DashboardGridLayoutWithAddCardPanel extends BaseComponent<
+            isOpen={isOpen}
+            cards={this.state.cardsForAddCardPanel}
+            moveCardFromAddCardPanelToDashboard={this._addCard}
+            onDismiss={this._onPanelDismiss}
+            draggingCardCallback={this._draggingCardCallback}
+            initialX={this.state.selectedCardInitialX}
+          />
+          <div className="dashboardContainerClassName">
+            <DashboardGridSectionLayout
+              isDraggable={isDraggable}
+              layout={this.state.layout}
+              sections={this.state.sections}
+              cards={this.state.dashboardCards}
+              dragApi={dragApi}
+              onLayoutChange={this._onLayoutChange}
+            />
+          </div>
+        </div>
       </>
     );
   }
 
+  private _draggingCardCallback = (
+    cardId: string,
+    title: string,
+    cardSize: CardSize,
+    initialX: number,
+    draggingAnimation?: DraggingAnimationType
+  ) => {
+    this.setState({
+      renderDraggingCard: true,
+      selectedCardId: cardId,
+      selectedCardInitialX: initialX,
+      selectedCardSize: cardSize,
+      selectedCardTitle: title,
+      draggingAnimation
+    });
+  };
+
+  private _hideDraggingCard = () => {
+    this.setState({
+      renderDraggingCard: false
+    });
+  };
+
   private _onPanelDismiss = () => {
     if (this.props.onPanelDismiss) {
       this.props.onPanelDismiss();
     }
   };
 
   private _onLayoutChange = (currentLayout: Layout[]): void => {
-    const newLayout: DashboardGridBreakpointLayouts = { lg: [] };
-    currentLayout.map((individualItemLayout: Layout) => {
-      const key: string = individualItemLayout.w.toString() + individualItemLayout.h.toString();
-      let cardSize = CardSize.small;
-      // recreating layout based off width and height of card. The width and height values are returned by RGl
-      // medium wide is 2 scale wide and 1 scale in height
-      // large is 2 scale high and wide
-      // mediumTall is 1 scale wide and 2 scales high
-      if (individualItemLayout.h === 1) {
-        cardSize = CardSize.section;
-      } else if (key === '28') {
-        cardSize = CardSize.large;
-      } else if (key === '18') {
-        cardSize = CardSize.mediumTall;
-      } else if (key === '24') {
-        cardSize = CardSize.mediumWide;
-      }
-      const itemLayout: IDashboardCardLayout = {
-        i: individualItemLayout.i!,
-        x: individualItemLayout.x,
-        y: individualItemLayout.y,
-        size: cardSize
-      };
-      newLayout.lg!.push(itemLayout);
-    });
-    if (newLayout !== this.state.layout) {
-      if (this.props.onLayoutChange) {
-        this.props.onLayoutChange(newLayout);
+    const lastCardIndex = currentLayout.length - 1;
+    // checking if a dragging card action is performed.
+    // If dragging is performed, dragging card is added to the layout whose id starts with 'n'
+    if (lastCardIndex > -1 && currentLayout[lastCardIndex].i!.startsWith('n')) {
```

Comment:
>  if (lastCardIndex > -1 && currentLayout[lastCardIndex].i!.startsWith('n')) { [](start = 2, length = 78)

I think we don't need this 'n' anymore. Can you see if we can do it without that.

Reply:
We need it as it differentiates a newly added card from the existing ones. We need this differentiation as the same method is called when there is a layout change caused by user moving cards around in the RGL.